src/syntax/InlineFootnoteReference.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include"../interfaces/ast.hpp"
+#include"../interfaces/parse.hpp"
+#include"../interfaces/syntax.hpp"
+#include"../utils.hpp"
+
+namespace almo{
+struct InlineFootnoteReference : public ASTNode{
+  private:
+    std::string symbol;
+    
+  public:
+    InlineFootnoteReference (std::string _symbol) : symbol(_symbol){
+        set_uuid();
+    }
+
+    std::string to_html() const override {
+        return "<span class=\"footnote-ref\"> <sup id=\"ref_" + symbol + "\"><a href=\"#note_" + symbol +"\">[" + symbol + "]</a></sup> </span>";
+    }
+
+    std::map<std::string, std::string> get_properties() const override {
+        return {
+            {"symbol", symbol}
+        };
+    }
+    std::string get_classname() const override {
+        return "InlineFootnoteReference";
+    }
+};
+struct InlineFootnoteReferenceSyntax : public InlineSyntax {
+    static inline const std::regex rex = std::regex(R"((.*?)\[\^(.*?)\](.*?))");
+    int operator()(const std::string &str) const override {
```

Comment:
ここの最後の正規表現がまずそうです. 最後に `(.*?)`　としてしまうと最短マッチになってしまうので、空文字列とマッチして残りの部分が全て捨てられてしまうと思います。 `?` は不要そうです

```suggestion
static inline const std::regex rex = std::regex(R"((.*?)\[\^(.*?)\](.*))");
```

Reply:
おそらく、

`static inline const std::regex rex = std::regex(R"(\[\^(.*?)\])");`

としてしまって 

```cpp
std::string prefix = sm.prefix().str();
std::string symbol = sm[1] ;
std::string suffix = sm.suffix().str();
```

とするのが一番よさそうです